modules/aggs-composite/src/main/java/org/elasticsearch/search/aggregations/composite/CompositeValuesSourceConfig.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.search.aggregations.composite;
+
+import org.elasticsearch.search.aggregations.support.ValuesSource;
+import org.elasticsearch.search.sort.SortOrder;
+
+class CompositeValuesSourceConfig {
+    private final ValuesSource vs;
+    private final int reverseMul;
+    private final boolean canEarlyTerminate;
+
+    CompositeValuesSourceConfig(ValuesSource vs, SortOrder order, boolean canEarlyTerminate) {
+        this.vs = vs;
+        this.canEarlyTerminate = canEarlyTerminate;
+        this.reverseMul = order == SortOrder.ASC ? 1 : -1;
+    }
+
+    ValuesSource valuesSource() {
+        return vs;
+    }
+
+    int reverseMul() {
```

Comment:
Could you add a javaDoc for this. It took me a while to work out what this actually was as it wasn't obvious from where its used